### DISCUSSION TITLE
Redirect to cart after failed payment

### DISCUSSION THREAD

**Discussion by benjibee** [duncanmcclean/simple-commerce#761] [Help]
I'm currently using **Mollie** in test mode, and trying to handle failed payments in a reasonable way.

Ideally I'd like to send a failed payment back to the payment method selection page with a flashed error message. Currently it seems that the error redirect can send the user to a given page, but the cart is by that time empty.

I've done a lot of digging around but can't quite track down where I could hook into some of these events to keep the cart and get the user back to where they chose their payment method with an error from the gateway.

If anyone out there has implemented this and has some thoughts on it I would very very keen to hear them!

**Comment by duncanmcclean** [comment]
Hey!

The cart certainly shouldn't be empty when you come back from Mollie after a failed payment - we should keep everything there. Can you please [open a bug report](https://github.com/duncanmcclean/simple-commerce/issues/new/choose) instead?
